src/vs/platform/url/common/urlIpc.ts

```diff
@@ -70,3 +72,37 @@ export class URLHandlerChannelClient implements IURLHandler {
 		return this.channel.call('handleURL', uri.toJSON());
 	}
 }
+
+export class URLHandlerRouter implements IClientRouter<string> {
+
+	constructor(private next: IClientRouter<string>) { }
+
+	async routeCall(hub: IConnectionHub<string>, command: string, arg?: any, cancellationToken?: CancellationToken): Promise<Client<string>> {
+		if (command !== 'handleURL') {
+			throw new Error(`Call not found: ${command}`);
+		}
+
+		if (arg) {
+			const uri = URI.revive(arg);
+
+			if (uri && uri.query) {
+				const match = /\bwindowId=([^&]+)/.exec(uri.query);
```

Comment:
Should this be more vscode specific, e.g. prefixed with vscode-windowId?

Reply:
That's up to you I guess. I would avoid `-` and use camelCase instead, though.

Reply:
@joaomoreno any concerns with using "-" in query params? I am doing that already in another place, so curious if there is a specific reason to avoid it.